Knuckledusters; New Ways to Beat People to Death in Style

TheKillfish opened the pull request:
## What Does This PR Do
Adds three flavours of knuckleduster and ways to get them. They are meant to be more concealable (ie. can fit in your pocket and have no hand sprite) bludgeoning instruments able to more easily break bone (or internal bleeding on slimes) when gripped harder (which also makes them nodrop, because good luck disarming someone gripping knuckledusters) and, if bones are broken, doing direct organ damage. Legality-wise, they are obviously weapons and thus can be confiscated at Security discretion if discovered during a search.

The following are the types of Knuckleduster, their stats, and any relevant additional info;
- **Knuckleduster**: The base type from which the other two are derived from. Deals 5 Brute damage and 3 when thrown, has a 10% chance to break bones on hit and causes 5 organ damage on broken parts, falls under the Small weight class, is Fireproof and is conductive (so no punching wires). Can be built at a hacked autolathe, costing 500 metal (which it can be recycled back into), and gives Combat Tech Level 2. In terms of raw damage, there are better and easier to get options, but these should still be servicable weapons between the nodrop, bone-breaking, organ damaging, and easy concealment
- **Syndicate Knuckleduster**: A fashionable black variant, avaliable to all Syndicate Operatives or rarely found as Maints loot. Compared to the normal knuckledusters, these are a large step up in lethality. They do 10 Brute damage and 5 when thrown, have a 15% chance to break bones on hit and cause 10 organ damage on broken parts, and additionally get the bonus of dealing an additional 5 organ damage regardless of bone breakage, and they give Illegal Tech Level 1. At the cost of 5 TC, they are meant to be a cheap but effective melee option for causing debilitation and damage less easily fixed than the stab wounds of the E-dagger, it's competitor for "cheap but dangerous weapons". As Contraband, they are intended to be C class, but can be upped to S class and removed from maints loot if balance team deems so.
- **Engraved Knuckleduster**: A golden variant engraved with the Nanotrasen initials that can be found in the Quartermaster's Locker on-station. Stats are mostly identical to the Syndicate Knuckleduster, but with the organ damage of the normal dusters and no bonus organ damage, along with being recyclable into Gold, giving Combat Tech Level 3, and having the bonus of being Acidproof. Similarly to the HOP and their Energy Pistol, QM's SOP should be updated to include a clause relating to this weapon; while permitted for self-defence, it is **not** a permission slip for doing Security's job, and can be confiscated if misused or abused. Stealing these as non-Command would constitute as Grand Theft.

## Why It's Good For The Game
More variation is always fun, and this gives players new alternatives with their upsides (capacity for nodrop, bon-breaking/IB and organ damage, pocketability) and downsides (not much raw damage, ranging from illegal to very illegal) compared to the usual suspects of Wooden Bats, Toolboxes and Spears. Having a weapon intended to break bones and damage organs will also shake combat up, since you can eventually punch hearts, lungs or brains into non-function and break arms or legs to hamper escape or armed resistance, with the Syndicate one being better at the former and the Engraved one the latter. The Engraved ones could even be made into a high-risk item for QM.

## Images of changes
![The Dusters 2](https://github.com/ParadiseSS13/Paradise/assets/145384199/46400c5f-8596-4b4b-809e-11c45e88957f)
The beautiful bois from left to right:
- Knuckleduster
- Syndicate Knuckleduster
- Engraved Knuckleduster

## Testing
Built, bought and found some, walked around with them in my pocket, beat Skrell and Slime People to death, fixed crashing issue with big help from Snowball on Discord, who provided fix.

## Changelog
:cl:
add: Added Knuckleduster to hacked Autolathe list, Syndicate Knuckleduster to Uplink and maints loot list, and the Engraved Knuckleduster to QM's station locker
/:cl:

TheKillfish commented:
Already decided to convert this into a draft, since I agree with folks on the Discord that this could do more, and I was going to originally but code was being disfunctional.

Plan for these;
- Make them more likely to break bones
- Add a nodrop functionality via `attack_self()`; strong for Syndie and weaker for the other two
- Find someone or somewhere better for the engraved one to go because I _very_ want someone in command to get this

Probablyjay19 commented:
>syndicate knuckledusters for just 5 TC 

![image](https://github.com/ParadiseSS13/Paradise/assets/87214315/374fc4cd-4ddc-4834-b83b-ece595ab44b4)

Lets gooooooo

Coolrune206 commented:
I don't see the NTR getting a purely lethals weapon roundstart 'cuz it feels like validbait. Additionally, while I see this as a generally nifty concealed improv/syndie weapon, knuckledusters don't really line up at all with the corporate feel of NanoTrasen, so I'm honestly not sure about these finding their way into Command. *Maybe* the quartermaster, for a bit of encouragement to his lackeys? But that's a pretty italicized maybe.

TheKillfish commented:
Yeah, I've decided to take it out of the NTR locker. I _do_ want it on-station somewhere (in a manner that won't need map changing preferrably), but more thought I give it, the more I realize it probably will just be an admeme thing, at least until it (hopefully) gets merged.

Christasmurf commented:
My first thought when I saw this was "quartermaster weapon"

Also the syndicate one looks pretty flat, could do with better shading like the others

TheKillfish commented:
Yeah, Syndie one is definitely the one I am least satisfied with, so I will be working on it once I've gotten code stuff done.

QM getting the engraved could work, maybe even potentially a High Risk item since QM (among others) lacks one, and Syndicate stealing the expensive prestige weapon for shits n giggles feeks like something they'd do.

Mitchs98 commented:
They'd definitely fit the QM quite well I feel, for the engraved dusters. And the QM *does* need more of a high risk item/Grand Theft item than their door remote

TheKillfish commented:
Dusters are now ready for review, with improved Syndicate Knuckleduster sprites (after much deliberation on how to make them look better), QM getting the Engraved ones instead of NTR, and new functions;
- Interact to tighten/relax grip on Knuckledusters, giving or removing nodrop.
- If gripped, you have a greater than normal chance to break bones and consequently do organ damage (to a random organ) if you punch parts with fractured bones (Slime People get IB instead of broken bones, because they don't deserve to escape the agony).
- Syndicate Knuckledusters can do direct organ damage (also to random organs) regardless of bones or not, so you can feasibly punch someone into deafness, blindness and brain death if you punch them in the head enough.

Mitchs98 commented:
Can't wait to go QM to knuckleduster some xenos.

Christasmurf commented:
If the engraved one is a "grand theft" item you may want to add it as a steal objective via code/game/gamemode/steal_items.dm

Protected job being the QM themselves obviously

Somniworld commented:
I'm a bit concerned by the ability of a weapon to randomly bonebreak someone. THis means that if the player is lucky enough, they can poke someone in the chest once, deal 5 damage, and somehow instantly fracture your ribs.

I feel like the dusters should deal more damage, keep the ability to reliably cause organ damage, but lose the bonebreak RNG element. Or perhaps you could tweak them so that their damage tresholds to cause fractures in the first place is lower, but, for short, RNG combat bad.

TheKillfish commented:
Tweaked the Syndie Knuckleduster bonus organ damage since DGamer mentioned on Discord that dealing 10-15 organ damage and potentially rupturing a lung immediately might be a balance concern (they said they'd DM someone on balance team, but I haven't heard back. Done this as posterity); now the Syndie dusters only deal 5 damage on broken bones and 5 unconditional bonus organ damage for a total of 5-10 organ damage.

I've also added a target health check to address Somni's concerns; as long as the target is over 90 health, bones cannot be fractured. I am not removing bone breaking, as it is interesting, but as a result it cannot be anything other than RNG. RNG combat is bad, but some RNG is fine. I have applied the same kind of check to the Syndie Duster's direct organ damage, so you can't potentially rupture someone's lung on the first hit. Of course, that all is assuming they're healthy when you start wailing on them, but imo it's fair game if they aren't.

Finally, I have added the Engraved Duster to the theft items list, so now all department heads have something a tot may need to steal.

Henri215 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,84 @@
+
+	var/obj/item/organ/external/punched = target.get_organ(user.zone_selected)
+	if(gripped && prob(robust) && target.health < 90) // Better at throwing strong punches when gripped
+		if(isslimeperson(target))
```
It would be better if you did check using TRAIT_NO_BONES, since your objective here is to affect those without bones and not simply slime people.

Henri215 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,84 @@
+		squishy.receive_damage(trauma) // Probably not so good for your organs to have your already broken ribs punched hard again
+		return
+
+
```
```suggestion
```

Henri215 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,84 @@
+	materials = list(MAT_GOLD = 500)
+	robust = 20
+
+/obj/item/melee/knuckleduster/test
```
This one was just create for testing? Going to remove it later?

TheKillfish replied:
I will probably give it a new root name, since I actually haven't been using it for testing and made it more as a funny admin weapon

Henri215 replied:
An admin version is really necessary? I doubt an admin would know that this item exists, and if one wishes for something similar he can always tweak the values by varediting it.

Henri215 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,84 @@
+		squishy.receive_damage(trauma)
+	if(punched.is_broken())
+		squishy.receive_damage(trauma) // Probably not so good for your organs to have your already broken ribs punched hard again
+		return
```
Is that return necessary?

TheKillfish replied:
Not really, done for posterity. I would assume it's fine without it if you're pointing it out

Henri215 replied:
Just to be sure, is your intent here that the organ could get receive_damage twice?

Henri215 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,84 @@
+	materials = list(MAT_METAL = 500)
+	origin_tech = "combat=2"
+	attack_verb = list("struck", "bludgeoned", "bashed", "smashed")
+	var/gripped = FALSE
+	var/elite = FALSE
+	var/robust = 10
+	var/trauma = 5
```
autodocument these vars please.

Henri215 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,83 @@
+	materials = list(MAT_METAL = 500)
+	origin_tech = "combat=2"
+	attack_verb = list("struck", "bludgeoned", "bashed", "smashed")
+	var/gripped = FALSE // Is the weapon gripped or not?
+	var/elite = FALSE // Can the weapon damage organs directly or not?
+	var/robust = 10 // Chance to break bones on hit
+	var/trauma = 5 // How much organ damage can the weapon do?
```
```suggestion
	/// Is the weapon gripped or not?
	var/gripped = FALSE
	/// Can the weapon damage organs directly or not?
	var/elite = FALSE
	/// Chance to break bones on hit
	var/robust = 10
	/// How much organ damage can the weapon do?
	var/trauma = 5
```
Now to turn this documentation into autodocs you do like this.

matttheficus commented:
Seems like a fun item, Balance side seems okay now. 

Added SoP and Law Update tags as itll need to be added to Grand Theft items + SoP for the QM being armed.

Sirryan2002 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,87 @@
+		flags &= ~NODROP
+		return
+
+/obj/item/melee/knuckleduster/attack(mob/living/target, mob/living/user, obj/item/organ/external, obj/item/organ/internal/)
```
```suggestion
/obj/item/melee/knuckleduster/attack(mob/living/target, mob/living/user)
```
unused parameters

Henri215 commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,87 @@
+	var/trauma = 5
+
+/obj/item/melee/knuckleduster/attack_self(mob/user)
+	if(!gripped)
+		gripped = TRUE
+		to_chat(user, "You tighten your grip on [src], ensuring you won't drop it.")
+		flags |= NODROP
+		return
+	if(gripped)
```
You can just do an if else here instead of 2 ifs.

DGamerL commented on code/game/objects/items/weapons/knuckledusters.dm:
```diff
@@ -0,0 +1,87 @@
+		return
+
+	var/obj/item/organ/internal/squishy = pick(punched.internal_organs)
+	if(gripped & elite && target.health < 90)
```
```suggestion
	if(gripped && elite && target.health < 90)
```

DGamerL commented on code/datums/uplink_items/uplink_general.dm:
```diff
@@ -374,6 +374,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
+	item = /obj/item/melee/knuckleduster/syndie
+	cost = 5
+	cant_discount = TRUE
+
```
```suggestion
```
